src/app/(frontend)/[id]/page.tsx

```diff
@@ -0,0 +1,92 @@
+"use client"
+import { useParams } from 'next/navigation';
+import React, { useState, useEffect } from 'react';
+import { Product } from '@/lib/models/ProductModel';
+import Image from 'next/image';
+import { register } from 'swiper/element/bundle';
+
+register();
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'swiper-container': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & {
+        navigation?: string;
+        pagination?: string;
+        loop?: string;
+      }, HTMLElement>,
+      'swiper-slide': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>,
+    }
+  }
+}
+
+const ProductDetailPage: React.FC = () => {
+  const [product, setProduct] = useState<Product | null>(null)
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const { id } = useParams()
+
+  useEffect(() => {
+    if (typeof id === 'string') {
+      setLoading(true);
+      fetch(`/api/${id}`)
+        .then((res) => {
+          if (res.ok) {
+            return res.json();
+          } else {
+            throw new Error('Failed to fetch product');
+          }
+        })
+        .then((data) => {
+          setProduct(data);
+          setLoading(false);
+        })
+        .catch((err) => {
+          setError(err.message || 'Failed to load product');
+          console.error('Fetch error:', err);
+          setLoading(false);
```

Comment:
Same reasoning with error:

```suggestion
          setProduct({ isLoading: false, error: err.message || 'Failed to load product' });
```

Reply:
Thanks for your tips and explanation Andres, whit this way we avoid unnecessary re-renders. Even though React is ready to support thousands of renders per second, I suppose these performance improvements should always be done, right?.

Reply:
@brunomembrado Exactly, maybe in this scenario having multiple re-renders isn't a big deal, but we should always prepare our application to have the best performance

Reply:
Thanks for the suggestion @andvirga. Makes sense, I will incorporate those changes now.